fairscale/optim/oss.py

```diff
@@ -389,17 +379,34 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
                 from a call to :meth:`state_dict`
         """
 
-        # NOTE: PyTorch 1.5 does not index linearly but with the id(params) at saving time
-        # we work around that here by using the fact that the params are ordered as in the param_groups
-        pytorch15_index_redirect = {k: i for i, k in enumerate(state_dict["state"].keys())}
```

Comment:
this was the arguably a little subtle bug: this implied that the loaded state dict and the param groups were sorted in the same way, which is not always the case actually (happened to be the case for the unit test in place..)